backend/src/main/java/org/eventplanner/events/ParticipationNotificationUseCase.java

```diff
@@ -0,0 +1,167 @@
+
+            event.setParticipationConfirmationsRequestsSent(alreadySentRequests + 1);
+            eventRepository.update(event);
+        });
+    }
+
+    private List<Event> getEventsToNotify(int alreadySentRequests) {
+        var currentYear = Instant.now().atZone(ZoneId.of("Europe/Berlin")).getYear();
+        var nextYear = currentYear + 1;
+
+        var getEvents = Stream.concat(
+                eventRepository.findAllByYear(currentYear).stream(),
+                eventRepository.findAllByYear(nextYear).stream()
+        ).toList();
+
+        return switch (alreadySentRequests) {
+            case 0 -> getEvents.stream()
+                    .filter(event -> event.getParticipationConfirmationsRequestsSent() == 0)
+                    // filter for 2 weeks remaining
+                    .filter(event -> event.getStart().minusWeeks(2).isBefore(Instant.now().atZone(ZoneId.of("Europe/Berlin"))))
+                    .toList();
+            case 1 -> getEvents.stream()
+                    .filter(event -> event.getParticipationConfirmationsRequestsSent() == 1)
+                    // filter for 1 week remaining
+                    .filter(event -> event.getStart().minusWeeks(1).isBefore(Instant.now().atZone(ZoneId.of("Europe/Berlin"))))
+                    .toList();
+            default -> List.of();
+        };
+    }
+
+    public void confirmRegistration(EventKey eventKey, RegistrationKey registrationKey, String accessKey) {
+        var event = eventRepository.findByKey(eventKey).orElseThrow();
+        var registration = event.getRegistrations().stream()
+                .filter(r -> registrationKey.equals(r.getKey()) && accessKey.equals(r.getAccessKey()))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException("Registration not found"));
+
+        if (registration.getConfirmedAt() != null) {
+            return;
+        }
+
+        registration.setConfirmedAt(Instant.now().atZone(ZoneId.of("Europe/Berlin")));
+        event.updateRegistration(registrationKey, registration);
+        eventRepository.update(event);
+    }
+
+    public void declineRegistration(EventKey eventKey, RegistrationKey registrationKey, String accessKey) {
+        // delete registration, find slot, delete registration key from slot, send email
+        var event = eventRepository.findByKey(eventKey).orElseThrow();
+        var registration = event.getRegistrations().stream()
+                .filter(r -> registrationKey.equals(r.getKey()) && accessKey.equals(r.getAccessKey()))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException("Registration not found"));
+
+        event.removeRegistration(registration.getKey());
+
+        var slot = event.getSlots().stream()
+                .filter(s -> registrationKey.equals(s.getAssignedRegistration()))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException("Slot not found"));
```

Comment:
if no slot exists, we should probably just ignore it and still save the event to remove the registration